src/services/utilities.ts

```diff
@@ -414,6 +414,57 @@ namespace ts {
             }
         }
     }
+    
+    export function isInString(sourceFile: SourceFile, position: number) {
+        let token = getTokenAtPosition(sourceFile, position);
+        return token && token.kind === SyntaxKind.StringLiteral && position > token.getStart();
+    }
+
+    export function isInComment(sourceFile: SourceFile, position: number) {
+        return isInCommentHelper(sourceFile, position, /*predicate*/ c => true);

```

Comment:
I would make the predicate optional.  And if it is not htere, then you don't check for it in the isInCommentHelper code.  This save an alloc here.